client/src/pages/components/Navbar.js

```diff
@@ -1,30 +1,57 @@
-import React from "react";
+import React, { useState }  from "react";
 import Logo from "../Logo/logo.png";
 import { Link } from "react-router-dom";
 import { FaSearch, FaUserCircle } from "react-icons/fa";
+import SearchEnergizers from "../components/SearchEnergizers";
 
-function Navbar() {
+
+function Navbar({ energizers }) {
+const [showSearch, setShowSearch] = useState(false);
+ const [query, setQuery] = useState("");
+
+const handleToggleSearch = () => {
+	setShowSearch(!showSearch);
+};
+
+ const handleSearch = (event) => {
+		setQuery(event.target.value);
+ };
     return (
-        <div>
-            <section className="logo-sec">
-                <img className="logo" src={Logo} alt="logo" />
-            </section>
-            <nav className="nav-bar">
-                <div className="search-wrapper nav-sub-wrapper">
-                    <FaSearch className="magnifier" />
-                    <input className="search-input" type="search" placeholder="Search..." />
-                </div>
-                <div className="pages-wrapper nav-sub-wrapper">
-                    <Link className="page" to="/home">Home</Link>
-                    <Link className="page create-energizer" to="/create-energizer">Create Energizer</Link>
-                    <Link className="page" to="/energizers">Energizers</Link>
-                </div>
-                <Link className="login-wrapper nav-sub-wrapper" to="/login">
-                    <FaUserCircle className="login-icon" />
-                    Log in</Link>
-            </nav>
-        </div>
-    );
-}
+			<div>
+				<section className="logo-sec">
+					<img className="logo" src={Logo} alt="logo" />
+				</section>
+				<nav className="nav-bar">
+					<div className="search-wrapper nav-sub-wrapper">
+						<FaSearch className="magnifier" onClick={handleToggleSearch} />
+						{showSearch && (
```

Comment:
This is very confusing, you are implementing the search functionality both inside this component <Navbar /> and then in the <SeachEnergizers /> component? Anyway you are referencing `energizers` but never passing any data into the <Navbar /> component so it will always be undefined.